### PR TITLE
os_name_in fix

### PR DESCRIPTION
**Pull request by zeitgeist**
`os_name_in` accepts illegal strings, and stores them under an illegal value.

Example test:
```SQL
create table a(os_name os_name);
insert into a values('unknonw'); -- notice the typo!
SELECT * FROM a ; -- => yields error
```

This PR is WIP; i.e. a proper migration & version number bump is missing; same goes for a backport for the `0.x` branch.

### CODE REVIEW

**Comment by zeitgeist** [comment]
will re-open against postgres 9.6 branch

**Comment by rapimo** [comment]
probably related to [this](https://github.com/adjust/pg-device_type/commit/1ee1b7dd76b4aecf268bc59689fa9e54efa7a81a)

**Comment by zeitgeist** [comment]
The fix is commit ee56446 which can also be ported / cherry-pick'd to the `postgres96` branch as-is.

Please review.

File: src/os_name.c [hunk 1]
```diff
@@ -1,6 +1,9 @@
 #include "os_name.h"
 PG_MODULE_MAGIC;
 
+static const os_name INVALID_OSNAME = 0;
```
Comment (rapimo) [inline src/os_name.c]: what's the rational defining that as const rather than using a macro?

File: src/os_name.c [hunk 2]
```diff
@@ -30,36 +33,36 @@ char *os_name_to_str(os_name c)
   }
 }
 
-static inline uint8
+static inline os_name
 check_os_name_num(const char *str, const char *expected, os_name os)
 {
     if (strcmp(expected, str) != 0)
-        return 0;
+        return INVALID_OSNAME;
```
Comment (rapimo) [inline src/os_name.c]: wouldn't it be less invasive to error out here right away
and or maybe make `INVALID_OSNAME()` a macro throwing that error?

```
#define INVALID_OSNAME(str) elog(ERROR, "unknown input os_name: %s", str)
```

File: src/os_name.c (hunk 1 above)
Comment (zeitgeist) [inline src/os_name.c]: no reason other than personal preference.

File: src/os_name.c (hunk 2 above)
Comment (zeitgeist) [inline src/os_name.c]: My motivation back then:
1. there are at least 12 calls to `check_os_name_num`, so we would have 12 places where we log something after the same pattern.
2. elog does not appear to be that lightweight: https://doxygen.postgresql.org/elog_8h_source.html#l00199

My assumption was, we only need 1 `elog` call in general, allowing the other string conversion helper functions to be inlined properly.

Comment (zeitgeist) [inline src/os_name.c]: if you prefer the elog macro version, I'll update the PR happily.

Comment (rapimo) [inline src/os_name.c]: do we assume that the function can't be inlined if it has an `elog` call ?

**Pull request closed without merging**